src/main/java/com/angelbroking/smartapi/http/SmartAPIRequestHandler.java

```diff
@@ -1,420 +1,379 @@
+        if (proxy != null) {
+            builder.proxy(proxy);
+        }
+
+        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
+        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
+        if (SmartConnect.enableLogging) {
+            client = builder.addInterceptor(logging).build();
+        } else {
+            client = builder.build();
+        }
+    }
+
+    public JSONObject apiHeaders() {
+
+        try {
+            JSONObject headers = new JSONObject();
+
+            // Local IP Address
+            InetAddress localHost = InetAddress.getLocalHost();
+            String clientLocalIP = localHost.getHostAddress();
+            headers.put(Constants.HEADER_CLIENT_LOCAL_IP, clientLocalIP);
+
+            // Public IP Address
+            String clientPublicIP = getPublicIPAddress();
+            headers.put(Constants.HEADER_CLIENT_PUBLIC_IP, clientPublicIP);
+            String macAddress = null;
+            // MAC Address
+            // get all network interfaces of the current system
+            Enumeration<NetworkInterface> networkInterface = NetworkInterface.getNetworkInterfaces();
+            // iterate over all interfaces
+            while (networkInterface.hasMoreElements()) {
+                // get an interface
+                NetworkInterface network = networkInterface.nextElement();
+                // get its hardware or mac address
+                byte[] macAddressBytes = network.getHardwareAddress();
+                if (macAddressBytes != null) {
+                    // initialize a string builder to hold mac address
+                    StringBuilder macAddressStr = new StringBuilder();
+                    // iterate over the bytes of mac address
+                    for (int i = 0; i < macAddressBytes.length; i++) {
+                        // convert byte to string in hexadecimal form
+                        macAddressStr.append(String.format("%02X%s", macAddressBytes[i], (i < macAddressBytes.length - 1) ? "-" : ""));
+                    }
+
+                    macAddress = macAddressStr.toString();
+                    if (macAddress != null) {
+                        break;
+                    }
+                }
+            }
+            headers.put(Constants.MAC_ADDRESS, macAddress);
+            String accept = "application/json";
+            headers.put(Constants.ACCEPT, accept);
+            String userType = "USER";
+            headers.put(Constants.USER_TYPE, userType);
+            String sourceID = "WEB";
+            headers.put(Constants.SOURCE_ID, sourceID);
+
+            logger.debug("Headers {}", headers);
+
+            return headers;
+        } catch (Exception e) {
+            logger.error(e.getMessage());
+            return null;
+        }
+
+    }
+
+    private String getPublicIPAddress() throws MalformedURLException {
+        String clientPublicIP = null;
+        URL urlName = new URL("http://checkip.amazonaws.com");

```

Comment:
Hardcoded URL?

Reply:
The URL http://checkip.amazonaws.com is a web-based IP address lookup service provided by Amazon Web Services (AWS). When a client requests this URL, the AWS service returns the public IP address of the client making the request.